### PR TITLE
Fixed the problem that the backslash could not be entered when the CRANE URL in the added cluster was entered normally

### PR DESCRIPTION
**Pull request by RuoChengF**
#### What type of PR is this?

Fix bugs from #769 

#### What this PR does / why we need it:

You cannot enter a backslash when entering a domain name

#### Which issue(s) this PR fixes:

Fixes #769

### CODE REVIEW

File: pkg/web/src/modules/editClusterSlice.ts [hunk 1]
```diff
@@ -1,3 +1,10 @@
+/*
+ * @Description: 
+ * @Author: 若城
+ * @Date: 2023-05-04 22:20:52
+ * @LastEditTime: 2023-05-08 09:27:05
+ */
+
```
Comment (qmhu) [inline pkg/web/src/modules/editClusterSlice.ts:7]: Please remove it

File: pkg/web/src/modules/editClusterSlice.ts [hunk 2]
```diff
@@ -61,12 +68,12 @@ const slice = createSlice({
     updateCluster: (state, action: PayloadAction<{ id: string; data: Partial<Omit<Cluster, 'id'>> }>) => {
       // Remove /
       // craneUrl = http://localhost:3000/ => http://localhost:3000
-      const { craneUrl } = action.payload.data;
-      if (craneUrl) {
-        action.payload.data.craneUrl = craneUrl.endsWith('/')
-          ? craneUrl.substring(0, craneUrl.lastIndexOf('/'))
-          : craneUrl;
-      }
+      // const { craneUrl } = action.payload.data;
```
Comment (qmhu) [inline pkg/web/src/modules/editClusterSlice.ts:71]: Not the correct way to implement it ,maybe try chatGPT ?

Comment (zsnmwy) [reply pkg/web/src/modules/editClusterSlice.ts:65]: If a backslash appears at the end of the URL, it needs to be removed.
If not the app will be displayed an error.

Please don't change it.

File: pkg/web/src/pages/Settings/cluster/EditClusterModal.tsx [hunk 3]
```diff
@@ -85,14 +85,24 @@ export const EditClusterModal = React.memo(() => {
   const validateCraneUrl = (id: string) => {
     const res = { error: false, msg: '' };
     const cluster = clusters.find((cluster: any) => cluster.id === id);
-
+    let reg = /^(?=^.{3,255}$)(http(s)?:\/\/)?(www\.)?[a-zA-Z0-9][-a-zA-Z0-9]{0,62}(\.[a-zA-Z0-9][-a-zA-Z0-9]{0,62})+(:\d+)*(\/\w+\.\w+)*$/;
```
Comment (zsnmwy) [inline pkg/web/src/pages/Settings/cluster/EditClusterModal.tsx:88]: The regex should cover more cases.

image

File: pkg/web/src/pages/Settings/cluster/EditClusterModal.tsx [hunk 4]
```diff
@@ -85,14 +85,24 @@ export const EditClusterModal = React.memo(() => {
...
+    
+// console.log(`cluster`, cluster);
```
Comment (zsnmwy) [inline pkg/web/src/pages/Settings/cluster/EditClusterModal.tsx:90]: Remove it

File: pkg/web/src/pages/Settings/cluster/EditClusterModal.tsx [hunk 5]
```diff
@@ -85,14 +85,24 @@ export const EditClusterModal = React.memo(() => {
...
       res.msg = t('Crane URL不能为空');
-    } else if (!cluster.craneUrl.startsWith('http://') && !cluster.craneUrl.startsWith('https://')) {
+    }else if(!reg.test(cluster.craneUrl)){
+      res.error = true;
+      res.msg = t('Crane URL格式不正确，请输入正确的URL');
+    }
+
+/**
+ *  原有校验方式
+ *    else if (!cluster.craneUrl.startsWith('http://') && !cluster.craneUrl.startsWith('https://')) {
```
Comment (zsnmwy) [inline pkg/web/src/pages/Settings/cluster/EditClusterModal.tsx:101]: Please remove the comment.
We can use Git to review the changes.
Don't need it.

File: pkg/web/src/modules/editClusterSlice.ts (hunk 1 above)
Comment (RuoChengF) [reply pkg/web/src/modules/editClusterSlice.ts:1]: ok

**Review by zsnmwy — CHANGES_REQUESTED** [review]

File: pkg/web/src/pages/Settings/cluster/EditClusterModal.tsx (hunk 4 above)
Comment (RuoChengF) [reply pkg/web/src/pages/Settings/cluster/EditClusterModal.tsx:90]: Whether you use the regex for validation

File: pkg/web/src/pages/Settings/cluster/EditClusterModal.tsx (hunk 3 above)
Comment (RuoChengF) [reply pkg/web/src/pages/Settings/cluster/EditClusterModal.tsx:88]: Do you use this regex for validation? The problem here is' substring(0, craneUrl.lastIndexOf('/')) '

File: pkg/web/src/modules/editClusterSlice.ts (hunk 2 above)
Comment (RuoChengF) [reply pkg/web/src/modules/editClusterSlice.ts:65]: Do you use this regex for validation? The problem here is' substring(0, craneUrl.lastIndexOf('/')) '

**Pull request closed without merging**